client/app/services/axios.js

```diff
@@ -1,26 +1,36 @@
 import axiosLib from "axios";
 import { Auth } from "@/services/auth";
 import qs from "query-string";
-import Cookies from "js-cookie";
+import createAuthRefreshInterceptor from "axios-auth-refresh";
 
 export const axios = axiosLib.create({
   paramsSerializer: params => qs.stringify(params),
+  xsrfCookieName: "csrf_token",
+  xsrfHeaderName: "X-CSRF-TOKEN",
 });
 
 const getData = ({ data }) => data;
 
 axios.interceptors.response.use(getData);
 
+createAuthRefreshInterceptor(
+  axios,
+  failedRequest => {
+    const message = failedRequest.response.data.message || "";
+    if (message.includes("CSRF")) {
+      return axios.get("/ping");
+    } else {
+      return Promise.reject();
+    }
+  },
+  { statusCodes: [400] }
+);
```

Comment:
Thinking about it further, I think this whole behavior needs to be somehow configurable. In the Databricks fork this behavior won't work/relevant but we would want to implement a different auth interceptor. 

Does the lib provide a way to remove an interceptor?

Reply:
Yes, you can easily eject interceptors.